### PR TITLE
RRD-73: Integrate draft with server

### PR DESCRIPTION
**Pull request by GigliSantiago**

### CODE REVIEW

File: src/api/index.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-import Config from 'react-native-config';
+// import Config from 'react-native-config';
 
 const BASE_SETTINGS = {
   timeout: 0,
   headers: {
     'content-type': 'application/json',
   },
-  baseURL: Config.APP_API_URL,
+  baseURL: 'http://localhost:3333',
```
Comment (julianv97) [inline src/api/index.ts]: Restore Config here

File: src/screens/draft/index.tsx [hunk 2]
```diff
@@ -45,95 +54,93 @@ const Draft = ({route}: TNavigation<Routes.DRAFT>) => {
...
-  );
-
   const listSeparator = () => <View style={styles.separator} />;
 
   const renderItemTeamA = (data: ListRenderItemInfo<IPlayer>) => {
     return (
-      <Text>
-        {data.item.firstName} {data.item.lastName}
-      </Text>
+      <View style={{width: '100%'}}>
```
Comment (julianv97) [inline src/screens/draft/index.tsx]: Move this inline style to the stylesheer

File: src/screens/draft/index.tsx [hunk 3]
```diff
@@ -45,95 +54,93 @@ const Draft = ({route}: TNavigation<Routes.DRAFT>) => {
...
+      </View>
     );
   };
 
   const renderItemTeamB = (data: ListRenderItemInfo<IPlayer>) => {
     return (
-      <Text>
-        {data.item.firstName} {data.item.lastName}
-      </Text>
+      <View style={{width: '100%'}}>
```
Comment (julianv97) [inline src/screens/draft/index.tsx]: Also here

**Review by julianv97 — CHANGES_REQUESTED** [review]

File: src/api/index.ts [hunk 4]
```diff
@@ -28,5 +28,7 @@ export const URL = {
     GET_INACTIVE_MATCHES: '/api/match/inactive',
     GET_ACTIVE_MATCHES: '/api/match/active',
     CREATE_MATCH: '/api/match',
+    RE_DRAFT: '/api/match/', //:id/draft
```
Comment (martinraveglia) [inline src/api/index.ts]: shouldn't this be as follows?

```suggestion
    RE_DRAFT: '/api/match/:id/draft',
```

File: src/helpers/skillChecker.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  if (skill >= 40 && skill < 60)
+    return <FeatherIcon size={25} color="#e4cc37" name="arrow-right" />;
+
+  if (skill >= 60 && skill < 80)
+    return <FeatherIcon size={25} color="#8fc93a" name="arrow-up-right" />;
+
+  if (skill >= 80 && skill < 100)
+    return <FeatherIcon size={25} color="#1e91d6" name="arrow-up" />;
+
+  return <FeatherIcon size={25} color="red" name="arrow-down" />;
```
Comment (martinraveglia) [inline src/helpers/skillChecker.tsx]:
```suggestion
  if (skill >= 80)
    return <FeatherIcon size={25} color="#1e91d6" name="arrow-up" />;

  if (skill >= 60)
    return <FeatherIcon size={25} color="#8fc93a" name="arrow-up-right" />;

  if (skill >= 40)
    return <FeatherIcon size={25} color="#e4cc37" name="arrow-right" />;

  if (skill >= 20)
    return <FeatherIcon size={25} color="#e18335" name="arrow-down-right" />;

  return <FeatherIcon size={25} color="red" name="arrow-down" />;
```

File: src/screens/draft/index.tsx [hunk 6]
```diff
@@ -45,95 +54,93 @@ const Draft = ({route}: TNavigation<Routes.DRAFT>) => {
...
+      </TouchableOpacity>
       <View style={styles.teamsContainer}>
-        <View style={styles.teamContainer}>
-          <Text style={styles.teamTitle}>Team A</Text>
-          <Text style={styles.averageSkill}>
-            Skill avg: {averageSkillTeamA}
+        <View style={styles.teamContainerA}>
+          <Text style={styles.teamTitleA}>Team A</Text>
+          <Text style={styles.averageSkillA}>
+            Skill avg: {match?.skillAvgA}
```
Comment (martinraveglia) [inline src/screens/draft/index.tsx]:
```suggestion
            Skill avg: {match?.skillAvgA ?? '--'}
```

File: src/screens/draft/index.tsx [hunk 7]
```diff
@@ -45,95 +54,93 @@ const Draft = ({route}: TNavigation<Routes.DRAFT>) => {
...
-          <Text style={styles.redraftContainer}>Re-draft</Text>
-        </TouchableOpacity>
-        <View style={styles.teamContainer}>
-          <Text style={styles.teamTitle}>Team B</Text>
-          <Text style={styles.averageSkill}>
-            Skill avg: {averageSkillTeamB}
+        <View style={styles.teamContainerB}>
+          <Text style={styles.teamTitleB}>Team B</Text>
+          <Text style={styles.averageSkillB}>
+            Skill avg: {match?.skillAvgB}
```
Comment (martinraveglia) [inline src/screens/draft/index.tsx]:
```suggestion
            Skill avg: {match?.skillAvgB ?? '--'}
```

File: src/screens/draft/index.tsx [hunk 8]
```diff
@@ -187,16 +192,18 @@ const Draft = ({route}: TNavigation<Routes.DRAFT>) => {
...
           valueInput={format(time, 'h:mm a')}
         />
-        <Button
-          isDisabled={false}
-          handleSubmit={handleSubmit(onSubmit)}
-          text="Confirmar partido"
-        />
+        <View style={styles.buttonContainer}>
+          <Button
+            customStyle={{width: '100%'}}
```
Comment (martinraveglia) [inline src/screens/draft/index.tsx]: Move this style to the stylesheet

File: src/screens/draft/index.tsx [hunk 9]
```diff
@@ -45,95 +54,93 @@ const Draft = ({route}: TNavigation<Routes.DRAFT>) => {
...
-      dataMatch.field +
-        '\n' +
-        dataMatch.date +
-        '\n' +
-        format(dataMatch.time, 'h:mm a'),
-      [{text: 'OK'}],
-    );
+
+    dispatch(updateMatch(dataMatch));
+    navigation.navigate(Routes.HOME);
```
Comment (martinraveglia) [inline src/screens/draft/index.tsx]: We are not handling potential errors from the dispatch. We should handle this case with a customToast (the component and use case is already done on the profile screen). If we encounter and error, the navigation to the Home Screen should not be done

File: src/screens/selectPlayers/index.tsx [hunk 10]
```diff
@@ -43,20 +47,26 @@ const SelectPlayers = ({
...
-    });
+    try {
+      const firebaseUID = getCurrentFirebaseUid();
+      if (!firebaseUID) throw new Error('FirebaseUID not found');
+      const payload: MatchBody = {
+        firebaseUID: firebaseUID,
+        players: selectedPlayers,
+      };
+      dispatch(createMatch(payload));
+      navigation.navigate(Routes.DRAFT);
```
Comment (martinraveglia) [inline src/screens/selectPlayers/index.tsx]: Add error handling!!

File: src/screens/selectPlayers/index.tsx [hunk 11]
```diff
@@ -80,7 +90,7 @@ const SelectPlayers = ({
         <Text style={styles.wideRowText}>
           {item?.firstName} {item?.lastName}
         </Text>
-        <Text style={styles.rowText}>{item.fidelity}</Text>
+        <Text style={styles.rowText}>{skillChecker(item.skill)}</Text>
```
Comment (martinraveglia) [inline src/screens/selectPlayers/index.tsx]: Why using a `<Text>` here if `skillChecker()` is returning an IconComponent? A plain `View` is more adequate

File: src/store/modules/match/selectors.ts [hunk 12]
```diff
@@ -1,8 +1,20 @@
...
 import {MatchStatus} from 'types/enums/match';
 import {RootState} from '../../store';
 
 export const matchData = (state: RootState) => state.match;
 
+export const historyMatchesData2 = createSelector([matchData], match => {
+  return match.matches.filter(item => {
+    return item.status !== MatchStatus.toBePlayed;
+  });
+});
```
Comment (martinraveglia) [inline src/store/modules/match/selectors.ts]: This is the same selector as `historyMatchesData`, remove it

File: src/store/modules/match/selectors.ts [hunk 13]
```diff
@@ -1,8 +1,20 @@
...
+
+export const currentMatch = createSelector(
+  [matchData],
+  match => match.currentMatch,
+);
```
Comment (martinraveglia) [inline src/store/modules/match/selectors.ts]: This is a simple selector, why using createSelector?
```suggestion
export const currentMatch = (state: RootState) => state.match.currentMatch
```

**Review by martinraveglia — CHANGES_REQUESTED** [review]

File: src/api/index.ts (hunk 4 above)
Comment (GigliSantiago) [reply src/api/index.ts]: A better solution might be,
```suggestion
    RE_DRAFT: (id: string) => '/api/match/${id}/draft',
```

**Review by julianv97 — APPROVED** [review]

File: src/screens/draft/index.tsx [hunk 14]
```diff
@@ -104,7 +111,35 @@ const Draft = ({navigation}: TNavigation<Routes.SELECT_PLAYERS>) => {
...
+            <CustomToast
+              id={id}
+              description={'Cannot create match'}
+              title={'Error'}
+              status={ToastStatus.error}
+            />
+          );
+        },
+      });
+    }
```
Comment (martinraveglia) [inline src/screens/draft/index.tsx]: Is this working? We are not waiting for the response to know if we are having an error.
Also, add the error to the console log please!

File: src/screens/selectPlayers/index.tsx [hunk 15]
```diff
@@ -55,9 +57,24 @@ const SelectPlayers = ({
...
+              <CustomToast
+                id={id}
+                description={'Cannot create match'}
+                title={'Error'}
+                status={ToastStatus.error}
+              />
+            );
+          },
+        });
+      }
```
Comment (martinraveglia) [inline src/screens/selectPlayers/index.tsx]: Same comment here!

**Review by martinraveglia — CHANGES_REQUESTED** [review]

**Review by martinraveglia — APPROVED** [review]

**Pull request merged**